posthog/queries/retention/retention_events_query.py

```diff
@@ -145,8 +145,11 @@ def get_query(self) -> Tuple[str, Dict[str, Any]]:
 
         null_person_filter = f"AND notEmpty({self.EVENT_TABLE_ALIAS}.person_id)" if self._using_person_on_events else ""
 
+        sample_clause = f"SAMPLE {self._filter.sampling_factor}" if self._filter.sampling_factor else ""
```

Comment:
Q: how does sampling work with groups? I guess it's better when sampling by distinctIDs for groups, since every event has a single group key. Huh, didn't realise that sampling by personID would bork groups instead.

Reply:
It's going to sample by distinct ID, so not great for groups but effectively random sampling I guess. 

Might bork conversions and the like

Reply:
Ah right, what am I saying, it's equally bad for both 😬 .